src/front-end/graphics.js

```diff
@@ -487,187 +530,199 @@ var Render = (function(){
-         myMatrix.translate(sx, sy);
-         myMatrix.rotate(rotation, 0, 0);
-         var parallelogram = s.path(path).transform(myMatrix.toTransformString())
-         parallelogram.data("name", properties.namepa)
-         var color = properties.colorpa;
-         parallelogram.attr({
+        var myMatrix = new Snap.Matrix();
+        myMatrix.translate(0, 0);
+        myMatrix.rotate(rotation, x, y);
+        if (style == "wedge") {
+          var f = s.rect(x,y - size ,size,size).transform(myMatrix.toTransformString());
+          f.attr({
+            fill: Utils.hex(0, 0, 205),
+            "fill-opacity": 0.6,
+          });
+          return f;
+        } else{
+          var prepMarkPath = [x,y-size,x+size,y-size,x+size,y]
+          var p = s.polyline(prepMarkPath).transform(myMatrix.toTransformString())
+          p.attr({
             fill: Utils.hex(color[0], color[1], color[2]),
-             "fill-opacity": color.a
-         });
-         parallelogram.drag(move, start, stop)
-     }
-
-
-
-    // helper method that draws bbox around an object
-    function _renderBoundingBox(s, obj) {
-        // render the bbox
-        var sq = s.path(obj.getBBox().path);
-        sq.attr({
             "fill-opacity": 0,
-            stroke: "#000",
-            strokeWidth: 1
-        })
-        return sq
-    }
+            stroke: Utils.hex(color[0], color[1], color[2]),
+            strokeWidth: 2
+          });
+          return p;
+        }
+      }
+
+      /**
+      * Renders a Regular Angle Mark on the canvas
+      * @param       {Snap} s  Snap.svg global object
+      * @param       {JSON} properties JSON object from Haskell server
+      */
+      function _renderRegularArc(s,properties){
+          var radius = properties.r
+          var color = properties.color
+          var x = properties.x
+          var y = properties.y
+          var style = properties.style
+          var angle = properties.angle > 360.0 ? 360.0
+                      - properties.angle : properties.angle //Handle angle > 360
+              angle = angle < 0 ? 360 + angle : angle //Handle angle < 0
+          var rotation = properties.rotation + 90 - angle
+
+          var startAngle = rotation
+          var endAngle = angle < 0 ? (rotation - angle) : (rotation + angle)
+
+          var arcPath = describeArc(x, y, radius, startAngle,endAngle,false);
+          var arc = s.path(arcPath);
+          arc.attr({
+              "fill-opacity": 0,
+              stroke: Utils.hex(color[0], color[1], color[2]),
+              strokeWidth: 2
+          });
+
+          if (style == "wedge") {
+              var pf = describeFullArc(x, y, radius, startAngle, endAngle);
+              var f = s.path(pf);
+              f.attr({
+                  // TODO: hardcoded color value here
+                  fill: Utils.hex(0, 0, 205),
+                  "fill-opacity": 0.6,
+              });
+              var g = s.g(arc,f)
+          } else{
```

Comment:
you can move the `var arcPath = ...; var arc = ...; arc.attr...` lines into the else case, so you don't do redundant computation in the frontend